test/parallel/test-zlib-bytes-read.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const zlib = require('zlib');
+
+const expectStr = 'abcdefghijklmnopqrstuvwxyz'.repeat(2);
+const expectBuf = Buffer.from(expectStr);
+
+function createWriter(target, buffer) {
+  const writer = { size: 0 };
+  const write = () => {
+    target.write(Buffer.from([buffer[writer.size++]]));
+    if (writer.size < buffer.length) {
+      setTimeout(write, 25);
```

Comment:
This looks a bit odd… I think what you want is to do something once the write finished, but the right way to do that would be to pass a callback to `target.write()`, instead of relying on timeouts